asmcli/asmcli.sh

```diff
@@ -2282,14 +2339,39 @@ configure_package() {
   fi
 
   if [[ "${USE_HUB_WIP}" -eq 1 ]]; then
-    kpt cfg set asm gcloud.project.environProjectID "${FLEET_ID}"
-    kpt cfg set asm anthos.servicemesh.hubMembershipID "${HUB_MEMBERSHIP_ID}"
-    kpt cfg set asm anthos.servicemesh.hub-idp-url "${HUB_IDP_URL}"
+    # VM installation uses the latest Hub WIP format
+    if [[ "${USE_VM}" -eq 1 ]]; then
+      kpt cfg set asm anthos.servicemesh.hubTrustDomain "${FLEET_ID}.svc.id.goog"
+      kpt cfg set asm anthos.servicemesh.hub-idp-url "${HUB_IDP_URL}"
+    # GKE-on-GCP installation uses legacy Hub WIP format to be consistent with GCP Hub public preview feature
+    else
+      kpt cfg set asm anthos.servicemesh.hubTrustDomain "${FLEET_ID}.hub.id.goog"
+      kpt cfg set asm anthos.servicemesh.hub-idp-url "https://gkehub.googleapis.com/projects/${FLEET_ID}/locations/global/memberships/${HUB_MEMBERSHIP_ID}"
+    fi
+  fi
+  if [[ -n "${CA_NAME}" && "${CA}" = "gcp_cas" ]]; then
+    kpt cfg set asm anthos.servicemesh.external_ca.ca_name "${CA_NAME}"
   fi
 
   if [[ "${USE_VM}" -eq 1 ]] && [[ "${_CI_NO_REVISION}" -eq 0 ]]; then
     kpt cfg set asm anthos.servicemesh.istiodHost "istiod-${REVISION_LABEL}.istio-system.svc"
     kpt cfg set asm anthos.servicemesh.istiodHostFQDN "istiod-${REVISION_LABEL}.istio-system.svc.cluster.local"
     kpt cfg set asm anthos.servicemesh.istiod-vs-name "istiod-vs-${REVISION_LABEL}"
   fi
+
+  if [[ "${CA}" == "mesh_ca" ]]; then
+    # set the trust domain aliases to include both new Hub WIP and old Hub WIP to achieve no downtime upgrade.
+    add_trust_domain_alias "${PROJECT_ID}.svc.id.goog"
+    add_trust_domain_alias "${PROJECT_ID}.hub.id.goog"
+    if [[ -n "${_CI_TRUSTED_GCP_PROJECTS}" ]]; then
+      # Gather the trust domain aliases from projects.
+      while IFS=',' read -r trusted_gcp_project; do
+        add_trust_domain_alias "${trusted_gcp_project}.svc.id.goog"
+      done <<EOF
+${_CI_TRUSTED_GCP_PROJECTS}
+EOF
+    fi
+    # shellcheck disable=SC2046
+    kpt cfg set asm anthos.servicemesh.trustDomainAliases $(context_get-option "TRUST_DOMAIN_ALIASES")
```

Comment:
should this be 
`run_command eval kpt cfg set asm anthos.servicemesh.trustDomainAliases "${TRUST_DOMAIN_ALIASES}"`

also, should this if block exist in `configure_ca` in `components/ca/mesh-ca.sh` instead of there?

Reply:
Yes and yes. However, I'm going to put those in another PR because the test isn't passing now since it requires yaml files that are in this commit. I'll commit this as-is and then add another PR with this fix and make sure tests pass.